Python/Product/PythonTools/PythonTools/LanguageServerClient/LspEditorUtilities.cs

```diff
@@ -22,10 +22,12 @@
 namespace Microsoft.PythonTools.LanguageServerClient {
     internal static class LspEditorUtilities {
         internal static SnapshotPoint GetSnapshotPositionFromProtocolPosition(this ITextSnapshot textSnapshot, Position position) {
-            var line = textSnapshot.GetLineNumberFromPosition(position.Line);
-            var snapshotPosition = textSnapshot.GetLineFromLineNumber(position.Line).Start + position.Character;
+            if (position.Line < textSnapshot.LineCount) {
```

Comment:
new code doesn't try to get the snapshot position if the line is out side of the snapshot.

now it will return the end position in this case